### PR TITLE
KEYCLOAK-7479: Sanitize potentially unsafe HTML

### PR DESCRIPTION
**Pull request by ssilvert**
Also see KEYCLOAK-7368

### CODE REVIEW

**Comment by ssilvert** [comment]
Several tests failing.  I'm not that surprised because the new sanitizer can alter html output.

**Comment by ssilvert** [comment]
All tests passing now.

File: distribution/feature-packs/server-feature-pack/pom.xml [hunk 1]
```diff
@@ -72,6 +72,16 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
```
Comment (stianst) [inline distribution/feature-packs/server-feature-pack/pom.xml]: Did you check if this dependency is available in Brew? If not we'll need to build it in Brew before we can merge so it doesn't break RH-SSO builds

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/MailUtils.java [hunk 2]
```diff
@@ -47,7 +47,13 @@ public static String getPasswordResetEmailLink(MimeMessage message) throws IOExc
 
     public static String getPasswordResetEmailLink(EmailBody body) throws IOException, MessagingException {
         final String textChangePwdUrl = getLink(body.getText());
-        final String htmlChangePwdUrl = getLink(body.getHtml());
+        String htmlChangePwdUrl = getLink(body.getHtml());
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/MailUtils.java]: Why is this needed? The link shouldn't be sanitized right?

File: testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractIdentityProviderTest.java [hunk 3]
```diff
@@ -357,7 +357,13 @@ protected String getVerificationEmailLink(MimeMessage message) throws IOExceptio
 
         final String htmlBody = (String) multipart.getBodyPart(1).getContent();
 
-        final String htmlChangePwdUrl = MailUtil.getLink(htmlBody);
+        String htmlChangePwdUrl = MailUtil.getLink(htmlBody);
```
Comment (stianst) [inline testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractIdentityProviderTest.java]: Why is this needed? The link shouldn't be sanitized right?

File: themes/src/main/resources/theme/base/account/account.ftl [hunk 4]
```diff
@@ -59,7 +59,7 @@
         <div class="form-group">
             <div id="kc-form-buttons" class="col-md-offset-2 col-md-10 submit">
                 <div class="">
-                    <#if url.referrerURI??><a href="${url.referrerURI}">${msg("backToApplication")?no_esc}/a></#if>
+                    <#if url.referrerURI??><a href="${url.referrerURI}">${kcSanitize(msg("backToApplication")?no_esc)}</a></#if>
```
Comment (stianst) [inline themes/src/main/resources/theme/base/account/account.ftl]: I take it there's no way to be able to do ?kc_sanitize instead?

Comment (ssilvert) [reply themes/src/main/resources/theme/base/account/account.ftl]: That's correct.  I wanted to do that but FreeMarker doesn't allow it.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/MailUtils.java (hunk 2 above)
Comment (stianst) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/MailUtils.java]: I don't have a problem with it being sanitized, but if it's not a valid link afterwards that doesn't seem correct

Comment (ssilvert) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/MailUtils.java]: It is a valid link.  I tried it to make sure.

File: distribution/feature-packs/server-feature-pack/pom.xml (hunk 1 above)
Comment (ssilvert) [reply distribution/feature-packs/server-feature-pack/pom.xml]: How do I check the Brew stuff?  I know nothing about Brew.

**Review by stianst — DISMISSED** [review]

**Review by stianst — DISMISSED** [review]

**Review by stianst — APPROVED** [review]
